django/contrib/gis/gdal/raster/source.py

```diff
@@ -57,6 +70,11 @@ def __init__(self, ds_input, write=False):
                         ds_input, err))
             else:
                 raise GDALException('Unable to read raster source input "{}"'.format(ds_input))
+        elif isinstance(ds_input, dict):
```

Comment:
This I am least sure about, I was trying to keep the input parameter to a single object, but baybe this should be changed to kwargs? The challenge is to "tell" the function if the input is a path to a file or a set of parameters, which is an either/or case.